elasticsearch/client.go

```diff
@@ -261,3 +262,31 @@ func (c Client) CreateIndex(i Indexable) (bool, error) {
 
 	return true, nil
 }
+
+func (c Client) DeleteIndex(i Indexable) error {
+	c.logger.Printf("Deleting index '%s' for %T", i.IndexName(), i)
+
+	endpoint := c.domain + "/" + i.IndexName()
+
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(i.IndexConfig()); err != nil {
+		return err
+	}
+	body := bytes.NewReader(buf.Bytes())
+
+	// Form the HTTP request
+	req, err := http.NewRequest(http.MethodDelete, endpoint, body)
+	if err != nil {
+		return err
+	}
+
+	// You can probably infer Content-Type programmatically, but here, we just say that it's JSON
```

Comment:
can you be the first person to stop copypasting these comments

Reply:
it might be nice to extract a private method that creates and signs the request, because it is done a lot

Reply:
I have removed the comments 🚀